lib/db_connection.ex

```diff
@@ -433,6 +433,38 @@ defmodule DBConnection do
   `pid` for connection crashes. So it is recommended to monitor the connected
   `pid` if you want to track all disconnections.
 
+  Here is an example of a `connection_listener` implementation:
+
+    defmodule DBConnectionListener do
+      use GenServer
+
+      def start_link() do
+        GenServer.start_link(__MODULE__, [], name: {:global, "db_connection_listener"})
+      end
+
+      @impl true
+      def init(stack) when is_list(stack) do
+        {:ok, stack}
+      end
+
+      @impl true
+      def handle_call(:read_state, _from, state) do
+        {:reply, state, state}
+      end
+
+      @impl true
+      def handle_info(msg, state) do
+        {:noreply, [msg | state]}
+      end
```

Comment:
Can we perhaps make this smarter? For example, handling `:connected` and `:disconnected` messages differently?

Reply:
Example: adding pids to the list on connect and removing them on disconnect

Reply:
Hi @v0idpwn! I actually went the opposite way and tried making it as dumb as possible. The goal was to have a  starting point on which folks could build on from.

Reply:
@v0idpwn, I added a function to match on the `:connected` state. Could you please expand on what you mean by "monitoring them and removing them on disconnect"? Sounds like a cool example! Please feel free to add commits to this if you'd like.